processor/k8sattributesprocessor/pod_association.go

```diff
@@ -32,44 +32,52 @@ import (
 func extractPodID(ctx context.Context, attrs pdata.Map, associations []kube.Association) (string, kube.PodIdentifier) {
 	// If pod association is not set
 	if len(associations) == 0 {
-		return extractPodIDNoAssociations(ctx, attrs)
+		_, id := extractPodIDNoAssociations(ctx, attrs)
+		return "", kube.PodIdentifier{
+			kube.PodIdentifierAttributeFromConnection(id),
+		}
 	}
 
 	connectionIP := getConnectionIP(ctx)
-	hostname := stringAttributeFromMap(attrs, conventions.AttributeHostName)
 	for _, asso := range associations {
-		// If association configured to take IP address from connection
-		switch {
-		case asso.From == "connection" && connectionIP != "":
-			return k8sIPLabelName, connectionIP
-		case asso.From == "resource_attribute":
-			// If association configured by resource_attribute
-			// In k8s environment, host.name label set to a pod IP address.
-			// If the value doesn't represent an IP address, we skip it.
-			if asso.Name == conventions.AttributeHostName {
-				if net.ParseIP(hostname) != nil {
-					return k8sIPLabelName, kube.PodIdentifier(hostname)
+		skip := false
+
+		ret := kube.PodIdentifier{}
+		for i, source := range asso.Sources {
+			// If association configured to take IP address from connection
+			switch {
+			case source.From == kube.ConnectionSource:
+				if connectionIP == "" {
+					skip = true
 				}
-			} else {
+				ret[i] = kube.PodIdentifierAttributeFromSource(source, connectionIP)
+			case source.From == kube.ResourceSource:
 				// Extract values based on configured resource_attribute.
-				attributeValue := stringAttributeFromMap(attrs, asso.Name)
-				if attributeValue != "" {
-					return asso.Name, kube.PodIdentifier(attributeValue)
+				attributeValue := stringAttributeFromMap(attrs, source.Name)
+				if attributeValue == "" {
+					skip = true
 				}
+
+				ret[i] = kube.PodIdentifierAttributeFromSource(source, attributeValue)
 			}
 		}
+
+		// If all association sources has been resolved, return result
+		if !skip {
+			return asso.Name, ret
```

Comment:
Isn't `asso.Name` deprecated? I'm not sure the first return argument even used anymore. Please update the function if it's not.